File: src/main/java/kz/superkassa/tests/framework/http/SuperkassaApiClient.java
package ru.superkassa.tests.framework.http;

import io.qameta.allure.restassured.AllureRestAssured;
import io.restassured.RestAssured;
import io.restassured.specification.RequestSpecification;
import ru.superkassa.tests.framework.config.TestConfig;

public final class SuperkassaApiClient {
    private final TestConfig config;

    public SuperkassaApiClient(TestConfig config) {
        this.config = config;
    }

    public RequestSpecification request() {
        return RestAssured.given()
                .filter(new AllureRestAssured())
                .baseUri(config.baseUrl())
                .header("Authorization", "Bearer " + config.apiToken())
                .contentType("application/json")
                .accept("application/json");
    }
}

File: src/main/kotlin/kz/superkassa/tests/framework/http/SuperkassaApiClient.kt
package kz.superkassa.tests.framework.http

import io.restassured.RestAssured
import io.restassured.specification.RequestSpecification
import kz.superkassa.tests.framework.config.TestConfig

@Suppress("unused")
class SuperkassaApiClient(
    private val config: TestConfig,
) {
    fun request(authPin: String = config.authPin): RequestSpecification =
        baseRequest()
            .header("Authorization", "Bearer $authPin")

    fun requestWithoutAuthorization(): RequestSpecification =
        baseRequest()

    private fun baseRequest(): RequestSpecification =
        RestAssured.given()
            .filter(AllureApiLoggingFilter())
            .baseUri(config.baseUrl)
            .contentType("application/json")
            .accept("application/json")
}
